internal/pkg/service/appproxy/http/router.go

```diff
@@ -80,6 +80,11 @@ func (r *Router) CreateHandler() http.Handler {
 	mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
 		appID, ok := parseAppID(req.Host)
```

Comment:
Host may be: `proxy.data-apps.svc.cluster.local`, you should improve the `parseAppID` function, and add some static expected host suffix to the proxy service configuration struct.